stack/scripts/multitenant_migrate.py

```diff
@@ -0,0 +1,701 @@
+                        action='store_true',
+                        default=False)
+
+    my_args = parser.parse_args(sys.argv[1:])
+
+    arg_vars = vars(my_args)
+
+    creds = arg_vars['super'].split(':')
+    if len(creds) != 2:
+        print('Superuser credentials not properly specified.  Must be "-u <user:pass>". Exiting...')
+        exit_on_error()
+    else:
+        arg_vars['superuser'] = creds[0]
+        arg_vars['superpass'] = creds[1]
+
+    creds = arg_vars['super'].split(':')
+    if len(creds) != 2:
+        print('Org admin credentials not properly specified.  Must be "-u <user:pass>". Exiting...')
+        exit_on_error()
+    else:
+        arg_vars['adminuser'] = creds[0]
+        arg_vars['adminpass'] = creds[1]
+
+    return arg_vars
+
+
+
+class Migrate:
+    def __init__(self):
+        self.args = parse_args()
+        self.start_date = self.args['date']
+        self.endpoint = self.args['endpoint']
+        self.metrics = {'reindex_start': '',
+                        'reindex_end': '',
+                        'appinfo_migration_start': '',
+                        'appinfo_migration_end': '',
+                        'full_data_migration_start': '',
+                        'full_data_migration_end': ''}
+        self.logger = init_logging(self.__class__.__name__)
+        self.super_user = self.args['superuser']
+        self.super_pass = self.args['superpass']
+        self.admin_user = self.args['adminuser']
+        self.admin_pass = self.args['adminpass']
+        self.org = self.args['org']
+        self.init = self.args['init']
+        self.full = self.args['full']
+
+    def run(self):
+        self.logger.info('Initializing...')
+
+        if not self.is_endpoint_available():
+            exit_on_error('Endpoint is not available, aborting')
+
+        if self.start_date is not None:
+            self.logger.info("Date Provided.  Re-index will run from date=[%s]", self.start_date)
+
+        try:
+
+            if self.full:
+
+                # Do full data migration and exit
+
+                self.start_fulldata_migration()
+
+                self.metrics['full_data_migration_start'] = get_current_time()
+                self.logger.info("Full Data Migration Started")
+                is_migrated = False
+                while not is_migrated:
+                    time.sleep(STATUS_INTERVAL_SECONDS)
+                    is_migrated = self.is_data_migrated()
+                    if is_migrated:
+                        break
+
+                self.metrics['full_data_migration_end'] = get_current_time()
+                self.logger.info("Full Data Migration completed")
+
+                self.log_metrics()
+                self.logger.info("Finished...")
+
+                return
+
+
+            if self.init:
+
+                # Init the migration system as this is the first migration done on the cluster
+
+                self.run_database_setup()
```

Comment:
I think if  you've already ran it manually at least once, then you're fine.  It sets up the default usergrid org, superuser acct, and the test org/account